ProcessMessage/handler.ts

```diff
@@ -384,6 +389,49 @@ export const getProcessMessageHandler = ({
 
               return result;
             }),
+            TE.chain(_ => {
```

Comment:
Consider move this sub-pipe into a dedicated function to increase redability and keep the main pipe at the same level of abstraction

Reply:
I'd also add a comment on what's happening

Reply:
Fixed in 8a9cfb5

Reply:
I've also added a comment in b2e90ed